### PR TITLE
glcontext_wgl: add OpenGLES support

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/glcontext_wgl.c [hunk 1]
```diff
@@ -184,15 +181,33 @@ static int wgl_init(struct glcontext *ctx, uintptr_t display, uintptr_t window,
...
+    if (ctx->backend == NGL_BACKEND_OPENGL) {
+        const int context_attributes[] = {
+            WGL_CONTEXT_MAJOR_VERSION_ARB, 1,
+            WGL_CONTEXT_MINOR_VERSION_ARB, 0,
+            WGL_CONTEXT_PROFILE_MASK_ARB, WGL_CONTEXT_CORE_PROFILE_BIT_ARB,
+            0
+        };
+
+        wgl->rendering_context = wgl->CreateContextAttribsARB(wgl->device_context, shared_context, context_attributes);
+    } else {
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext_wgl.c:195]: `if (ctx->backend == NGL_BACKEND_OPENGL) {` + `ngli_assert(0)` in the `else` case?

File: libnodegl/glcontext_wgl.c [hunk 2]
```diff
@@ -184,15 +181,33 @@ static int wgl_init(struct glcontext *ctx, uintptr_t display, uintptr_t window,
...
-        WGL_CONTEXT_MINOR_VERSION_ARB, 0,
-        WGL_CONTEXT_PROFILE_MASK_ARB, WGL_CONTEXT_CORE_PROFILE_BIT_ARB,
-        0
-    };
-
     HGLRC shared_context = (HGLRC)other;
-    wgl->rendering_context = wgl->CreateContextAttribsARB(wgl->device_context, shared_context, context_attributes);
+
+    if (ctx->backend == NGL_BACKEND_OPENGL) {
+        const int context_attributes[] = {
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext_wgl.c:187]: `static`?

File: libnodegl/glcontext_wgl.c [hunk 3]
```diff
@@ -184,15 +181,33 @@ static int wgl_init(struct glcontext *ctx, uintptr_t display, uintptr_t window,
...
+
+        wgl->rendering_context = wgl->CreateContextAttribsARB(wgl->device_context, shared_context, context_attributes);
+    } else {
+        const char *extensions = wgl->GetExtensionsStringARB(wgl->device_context);
+        if (!ngli_glcontext_check_extension("WGL_EXT_create_context_es2_profile", extensions) &&
+            !ngli_glcontext_check_extension("WGL_EXT_create_context_es_profile", extensions)) {
+            LOG(ERROR, "OpenGLES is not supported by this device");
+            return -1;
+        }
+        const int context_attributes[] = {
```
Comment (cboesch-gpsw) [inline libnodegl/glcontext_wgl.c:202]: ditto `static`

**Review by cboesch-gpsw — APPROVED** [review]
LGTM aside from the cosmetics

File: libnodegl/glcontext_wgl.c (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c:202]: Done.

File: libnodegl/glcontext_wgl.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c]: Done.

File: libnodegl/glcontext_wgl.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/glcontext_wgl.c]: Done.

**Pull request merged**